### DISCUSSION TITLE
Need information for pihole plugin

### DISCUSSION THREAD

**Discussion by pascalbourges** [netalertx/NetAlertX#806] [General]
Hi,
What is the table used by pihole plugin to map name with IP ? 
I have populated /etc/pihole/custom.list file to map IPs with my personnal names, i have put this (with python script)  in network_addresses table, but in NetAlertX i have not my personnal names.. 
Regards

**Comment by jokob-sk** [comment]
Hi @pascalbourges, 

The pihole plugin is pretty straightforward - it's a DB mapping script - the SQL query can be found and modified (caution is advised) in the `PIHOLE_CMD`  setting.

The default value is:

```
SELECT n.hwaddr AS Object_PrimaryID, {s-quote}null{s-quote} AS Object_SecondaryID, datetime() AS DateTime, na.ip  AS Watched_Value1, n.lastQuery AS Watched_Value2, na.name AS Watched_Value3, n.macVendor AS Watched_Value4, {s-quote}null{s-quote} AS Extra, n.hwaddr AS ForeignKey FROM EXTERNAL_PIHOLE.Network AS n LEFT JOIN EXTERNAL_PIHOLE.Network_Addresses AS na ON na.network_id = n.id WHERE n.hwaddr NOT LIKE {s-quote}ip-%{s-quote} AND n.hwaddr is not {s-quote}00:00:00:00:00:00{s-quote}  AND na.ip is not null
```

Hope this helps,
j

**Reply by pascalbourges** [reply]
Thanks a lot, i understand better. I'll see how to modify code for me. 
Best regards
Pascal
